core/services/keeper/integration_test.go

```diff
@@ -0,0 +1,118 @@
+	"github.com/stretchr/testify/require"
+)
+
+var oneEth = big.NewInt(1000000000000000000)
+var oneHunEth = big.NewInt(0).Mul(oneEth, big.NewInt(100))
+
+func TestKeeperEthIntegration(t *testing.T) {
+	g := gomega.NewGomegaWithT(t)
+
+	// setup node key
+	nodeKey := cltest.MustGenerateRandomKey(t)
+	nodeAddress := nodeKey.Address.Address()
+	nodeAddressEIP55, err := models.EIP55AddressFromAddress(nodeAddress)
+	require.NoError(t, err)
+
+	// setup blockchain
+	sergey := cltest.NewSimulatedBackendIdentity(t) // owns all the link
+	steve := cltest.NewSimulatedBackendIdentity(t)  // registry owner
+	carrol := cltest.NewSimulatedBackendIdentity(t) // client
+	nelly := cltest.NewSimulatedBackendIdentity(t)  // other keeper operator
+	genesisData := core.GenesisAlloc{
+		sergey.From: {Balance: oneEth},
+		steve.From:  {Balance: oneEth},
+		carrol.From: {Balance: oneEth},
+		nelly.From:  {Balance: oneEth},
+		nodeAddress: {Balance: oneEth},
+	}
+
+	gasLimit := goEthereumEth.DefaultConfig.Miner.GasCeil * 2
+	backend := backends.NewSimulatedBackend(genesisData, gasLimit)
+
+	payload1 := common.Hex2Bytes("1234")
+	payload2 := common.Hex2Bytes("ABCD")
+
+	linkAddr, _, linkToken, err := link_token_interface.DeployLinkToken(sergey, backend)
+	require.NoError(t, err)
+	gasFeedAddr, _, _, err := mock_v3_aggregator_contract.DeployMockV3AggregatorContract(steve, backend, 18, big.NewInt(60000000000))
+	require.NoError(t, err)
+	linkFeedAddr, _, _, err := mock_v3_aggregator_contract.DeployMockV3AggregatorContract(steve, backend, 18, big.NewInt(20000000000000000))
+	require.NoError(t, err)
+	regAddr, _, registryContract, err := keeper_registry_wrapper.DeployKeeperRegistry(steve, backend, linkAddr, linkFeedAddr, gasFeedAddr, 250_000_000, big.NewInt(1), 20_000_000, big.NewInt(3600), big.NewInt(60000000000), big.NewInt(20000000000000000))
+	require.NoError(t, err)
+	upkeepAddr, _, upkeepContract, err := basic_upkeep_contract.DeployBasicUpkeepContract(carrol, backend)
+	require.NoError(t, err)
+	_, err = linkToken.Transfer(sergey, carrol.From, oneHunEth)
+	require.NoError(t, err)
+	_, err = linkToken.Approve(carrol, regAddr, oneHunEth)
+	require.NoError(t, err)
+	_, err = registryContract.SetKeepers(steve, []common.Address{nodeAddress, nelly.From}, []common.Address{nodeAddress, nelly.From})
+	require.NoError(t, err)
+	_, err = registryContract.RegisterUpkeep(steve, upkeepAddr, 2_500_000, carrol.From, []byte{})
+	require.NoError(t, err)
+	_, err = upkeepContract.SetBytesToSend(carrol, payload1)
+	require.NoError(t, err)
+	_, err = upkeepContract.SetShouldPerformUpkeep(carrol, true)
+	require.NoError(t, err)
+	backend.Commit()
+	_, err = registryContract.AddFunds(carrol, big.NewInt(0), oneHunEth)
+	require.NoError(t, err)
+	backend.Commit()
+
+	stopMining := cltest.Mine(backend)
+	defer stopMining()
+
+	// setup app
+	config, _, cfgCleanup := cltest.BootstrapThrowawayORM(t, "keeper_eth_integration", true, true)
```

Comment:
I may be mistaken here, but I believe I needed this because the spawner uses a postgres callback to start new services, and this isn't triggered in the transactional tests. I believe the same goes for the BPTXM.